src/analysis/convex/basic.lean

```diff
@@ -479,11 +479,49 @@ def convex_on (s : set E) (f : E → β) : Prop :=
   ∀ ⦃x y : E⦄, x ∈ s → y ∈ s → ∀ ⦃a b : ℝ⦄, 0 ≤ a → 0 ≤ b → a + b = 1 →
     f (a • x + b • y) ≤ a • f x + b • f y
 
+/-- Concavity of functions -/
+def concave_on (s : set E) (f : E → β) : Prop :=
+  convex s ∧
+  ∀ ⦃x y : E⦄, x ∈ s → y ∈ s → ∀ ⦃a b : ℝ⦄, 0 ≤ a → 0 ≤ b → a + b = 1 →
+    a • f x + b • f y ≤ f (a • x + b • y)
+
+/-- A function f is concave iff -f is convex -/
+lemma concave_on_iff_neg_convex_on {γ : Type*} [ordered_add_comm_group γ] [ordered_semimodule ℝ γ]
+  (s : set E) (f : E → γ) : concave_on s f ↔ convex_on s (-f) :=
```

Comment:
If you swap the sides of the `iff`, this can become a simp lemma (and please change the name accordingly, it could become `neg_convex_on_iff`).

Reply:
Done